src/App.js

```diff
@@ -1,60 +1,62 @@
-import { ApolloProvider } from "@apollo/react-hooks"
-import ApolloClient from "apollo-boost"
-import PrivateRoute from "./PrivateRoute"
-import Auth0 from "./Auth0"
-import Home from "./views/Home"
-import OrgTopLevel from "./views/Organization"
-import PdfGenerator from "./views/Labels/PdfGenerator"
-import Labels from "./views/Labels/Labels"
+import React, { useEffect, useState } from "react";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
+import { ApolloProvider } from "@apollo/react-hooks";
+import ApolloClient from "apollo-boost";
+import PrivateRoute from "./PrivateRoute";
+import Auth0 from "./Auth0";
+import Home from "./views/Home";
+import OrgTopLevel from "./views/Organization";
+import PdfGenerator from "./views/Labels/PdfGenerator";
+import Labels from "./views/Labels/Labels";
 
-const { REACT_APP_GRAPHQL_SERVER } = process.env
-
-const client = new ApolloClient({
-  uri: REACT_APP_GRAPHQL_SERVER,
-})
+const { REACT_APP_GRAPHQL_SERVER } = process.env;
 
 export default function App() {
-  const [loggedIn, setLoggedIn] = useState(false)
-  const [authObject, setAuthObject] = useState({})
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [authObject, setAuthObject] = useState({});
+
+  const client = new ApolloClient({
+    uri: REACT_APP_GRAPHQL_SERVER,
+    request: (operation) => {
+      operation.setContext({
+        headers: {
+          Authorization: `Bearer ${authObject.accessToken}`,
+          "X-Boxwise": authObject.idToken,
+          "X-Clacks-Overhead": "GNU Terry Pratchett",
+        },
+      });
+    },
+  });
 
   function handleLogIn() {
-    // Auth0 returns the token in the params of the URL after successful login and redirect
-    const totalHash = window.location.hash
-    if (totalHash) {
-      // remove leading # and split into components,
-      // so now you have ['key1=value1', 'key2=value2']
-      const hashArray = totalHash.substr(1).split("&")
-      const authObject = {}
-      hashArray.forEach((item) => {
-        const keyValArray = item.split("=")
-        // turns [key, value] into authObject={key: value}
-        // eslint-disable-next-line prefer-destructuring
-        authObject[keyValArray[0]] = keyValArray[1]
+    // if the login was successful, there will be a hash in the url, so you can do all the parsing work in the Auth0 file
+    Auth0.handleAuthentication()
+      .then((authTokens) => {
+        setAuthObject(authTokens);
+        !!authTokens ? setLoggedIn(true) : setLoggedIn(false);
       })
-      // the auth object has many items in it, we generally care about the access_token
-      setAuthObject(authObject)
-      // this is where you would also handle authorization errors from the API
-      // eslint-disable-next-line no-unused-expressions
-      authObject.access_token ? setLoggedIn(true) : setLoggedIn(false)
-    }
+      .catch((err) => {
+        // TODO: better logging and error handling
+        console.log(err);
```

Comment:
I guess the plan is to put in Sentry again or similar?

Reply:
@mcgnly yes we should put in Sentry again

Reply:
out of scope for this PR though, I would say

Reply:
@mcgnly yes, absolutely :)